.circleci/config.yml

```diff
@@ -1,240 +1,312 @@
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Client.Legacy.Test/Client.Legacy.Test.csproj
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Client.Linq.Test/Client.Linq.Test.csproj
-          sed -i '/<TargetFrameworks>netcoreapp3.1;net5.0;net6.0;net7.0<\/TargetFrameworks>/c\<TargetFramework>'net7.0'<\/TargetFramework>' Examples/Examples.csproj
-      - run:
-          name: Check compilation warnings
-          command: |
-            dotnet clean --configuration Release
-            dotnet build --configuration Release -warnAsError -warnAsMessage:CS0618,NETSDK1138
-      
+      - attach_workspace:
+          at: .
+      - influxdb_install_windows:
+          zip-file-name: influxdb2-2.7.5-windows.zip
+          influxdb-version: 2.7.5
+      - influxdb_install_windows:
+          zip-file-name: influxdb-1.8.10_windows_amd64.zip
+          influxdb-version: 1.8.10
+      - influxdb_run_windows:
+          influxdb-version: 2.7.5
+          influxdb-port: 9999
+          influxdb-data-dir: ~/.influxdbv2
+      - influxdb_run_windows:
+          influxdb-version: 2.7.5
+          influxdb-port: 8888
+          influxdb-data-dir: ~/.influxdbv2-onboarding-tests
+      - influxdb_run_windows:
+          influxdb-version: 1.8.10
+          influxdb-port: 8086
+          influxdb-data-dir: ~/.influxdbv1
+      - dotnet_info
+      - test_influxdb_client:
+          dotnet-target-framework: << parameters.dotnet-target-framework >>
+          code-coverage-report: << parameters.code-coverage-report >>
+
   check-code-formatting:
     docker:
       - image: *default-dotnet-image
+    working_directory: ~/influxdb-client-csharp
     steps:
       - checkout
       - restore_cache:
-          name: Restoring reSharper Cache
+          name: "Restore ReSharper cache"
           keys:
-              - &cache-key reSharper-cache-2022_1_0
+            - &cache-key ReSharper-cache-2022_1_0
       - run:
-          name: Check code formatting
+          name: "Check code formatting"
+          shell: bash
           command: |
             ./Scripts/reformat-code.sh
             git diff --exit-code
       - save_cache:
-            name: Saving reSharper Cache
-            key: *cache-key
-            paths:
-                - ./reSharperCLI
-  
+          name: "Save ReSharper cache"
+          key: *cache-key
+          paths:
+            - ./ReSharperCLI
+
   deploy-preview:
-    parameters:
-      influxdb-image:
-        type: string
-        default: *default-influxdb-image
     docker:
       - image: *default-dotnet-image
+    working_directory: ~/influxdb-client-csharp
     steps:
       - run:
           name: Early return if this build is from a forked repository
           command: |
             if [[ $CIRCLE_PROJECT_USERNAME != "influxdata" ]]; then
```

Comment:
I'm not sure that this condition is useful because the environment variable for the Bonitoo API key (`BONITOO_SNAPSHOT_APIKEY`) is only available when executed within the InfluxData CircleCI account.
It is also possible for anyone to open a PR and change this file to bypass it 😕

Reply:
Due to security and privacy considerations, certain environment variables like `BONITOO_SNAPSHOT_APIKEY` are not automatically propagated to forked repositories. This is a common practice in CI/CD environments to prevent unauthorized access to sensitive information or resources that the environment variable might grant access to.

As a result, when the CI/CD pipeline runs on a forked repository without access to this environment variable, any steps or conditions in the pipeline that rely on `BONITOO_SNAPSHOT_APIKEY` will not execute as intended. This mechanism effectively prevents the failure of the forked repository's pipeline that would occur due to the absence of required credentials or API keys.

Reply:
Of course it's completely normal that `BONITOO_SNAPSHOT_APIKEY` is only available to the InfluxData CircleCI pipeline and not propagated to forked repositories.

I actually thought this condition was meant to prevent publishing NuGet packages when it was not from the InfluxData pipeline (which is already the case since `BONITOO_SNAPSHOT_APIKEY` is not specified), but it is rather to avoid failure for forked repositories.
Thanks for this explanation 😄